go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -107,6 +107,12 @@ func (ep *TabletPlan) buildAuthorized() {
 	}
 }
 
+var (
+	// Global stats vars.
+	// TODO(sougou): unglobalize after componentizing TabletServer.
+	queryCounts, queryTimes, queryRowCounts, queryErrorCounts *stats.CountersWithMultiLabels
```

Comment:
Shouldn't these just stay as part of the QueryEngine?

Reply:
Putting them inside QueryEngine causes panics in tests when code tries to update these vars, which get initialized only for the first instance.

We can move these in once we componentize it, which is what the TODO is for.